### PR TITLE
GEN-27

### PR DESCRIPTION
**Pull request by jazzplato**
refactored the `TestServer` from the `run_junit.py` into `util`
- created a `TestServer` class for the common functionalities
- created a `TestJUnit` class for running JUnit tests
- created a `constants` file to consolidate all the magic numbers/strings
- refactored the `run_junit.py` to use the `TestJUnit` class

### CODE REVIEW

File: script/testing/util/test_junit.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+from util import constants
+from util.test_server import TestServer
+
+
+class TestJUnit(TestServer):
+    """ Class to run JUnit tests """
+    def __init__(self, args):
+        TestServer.__init__(self, args)
+        self.test_command = constants.JUNIT_TEST_COMMAND
+        self.test_output_file = constants.JUNIT_DB_OUTPUT_FILE
```
Comment (bialesdaniel) [inline script/testing/util/test_junit.py]: Do we need to parse the args. For example, getting the output file from the args to override the default?

File: script/testing/util/test_server.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+        self.set_path()
+        self.db_process = None
+
+        # db server location
+        self.db_host = constants.DEFAULT_DB_HOST
+        self.db_port = constants.DEFAULT_DB_PORT
+
+        # test execution output
+        self.test_output_file = ""
```
Comment (bialesdaniel) [inline script/testing/util/test_server.py]: Do you think its benefitial to designate some default here (maybe based on the time+date so it doesn't create naming conflicts)

File: script/testing/util/test_server.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+        # server output
+        self.db_output_file = constants.DEFAULT_DB_OUTPUT_FILE
+
+        self.set_path()
+        self.db_process = None
+
+        # db server location
+        self.db_host = constants.DEFAULT_DB_HOST
+        self.db_port = constants.DEFAULT_DB_PORT
```
Comment (bialesdaniel) [inline script/testing/util/test_server.py]: The user needs a way to overwrite these. Do we need to parse the `args` or is that handled elsewhere?

File: script/testing/junit/run_junit.py [hunk 4]
```diff
@@ -1,138 +1,14 @@
...
-        self._run_db_server()
-        ret_val = self._run_junit()
-        self._print_output(self.junit_output_file)
-
-        self._stop_db_server()
-        if ret_val:
-            # print the db_server log file, only if we had a failure
-            self._print_output(self.db_server_output_file)
-        return ret_val
+from util.test_junit import TestJUnit
```
Comment (benliangw) [inline script/testing/junit/run_junit.py]: Do we need to put all imports at the top of the file? [ref](https://www.python.org/dev/peps/pep-0008/#imports)

Comment (benliangw) [inline script/testing/junit/run_junit.py]: Do we need to put all imports at the top of the file? [ref](https://www.python.org/dev/peps/pep-0008/#imports)

File: script/testing/util/test_server.py (hunk 3 above)
Comment (benliangw) [reply script/testing/util/test_server.py]: +db_output_file

File: script/testing/util/test_junit.py (hunk 1 above)
Comment (jazzplato) [reply script/testing/util/test_junit.py]: this has been done in the recent commits

File: script/testing/util/test_server.py (hunk 3 above)
Comment (jazzplato) [reply script/testing/util/test_server.py]: @bialesdaniel these 2 have been handled
@benliangw this one not yet

File: script/testing/junit/run_junit.py (hunk 4 above)
Comment (jazzplato) [reply script/testing/junit/run_junit.py]: if you are talking about wildcard importing like `from ABC import *` then the answer is no. [PEP8](https://www.python.org/dev/peps/pep-0008/#imports) mentions that 
> Wildcard imports (from <module> import *) should be avoided

Comment (jazzplato) [reply script/testing/junit/run_junit.py]: oh i see your point. the following 2 lines are **essential** for you to import the `TestJUnit` successfully due to [sys.path of the main process](https://chrisyeh96.github.io/2017/08/08/definitive-guide-python-imports.html#more-on-syspath)
```python
base_path = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
sys.path.insert(0, base_path)
```

File: script/testing/util/test_server.py (hunk 2 above)
Comment (jazzplato) [reply script/testing/util/test_server.py]: done in the recent commits

File: script/testing/util/test_server.py (hunk 3 above)
Comment (jazzplato) [reply script/testing/util/test_server.py]: @benliangw now fixed

**Pull request merged**